src/transformers/generation/flax_utils.py

```diff
@@ -389,7 +389,7 @@ def generate(
                 UserWarning,
             )
         elif generation_config.max_new_tokens is not None:
-            if not has_default_max_length:
+            if not has_default_max_length and generation_config.max_length is not None:
```

Comment:
The logic in this line could be simplified to `if kwargs.get("max_length") is not None:`, but I believe this way is much easier to understand the intent behind the warning :)